pkg/cri/opts/spec_windows.go

```diff
@@ -0,0 +1,121 @@
+	"golang.org/x/sys/windows"
+	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
+)
+
+func escapeAndCombineArgsWindows(args []string) string {
+	escaped := make([]string, len(args))
+	for i, a := range args {
+		escaped[i] = windows.EscapeArg(a)
+	}
+	return strings.Join(escaped, " ")
+}
+
+// WithProcessCommandLineOrArgsForWindows sets the process command line or process args on the spec based on the image
+// and runtime config
+// If image.ArgsEscaped field is set, this function sets the process command line and if not, it sets the
+// process args field
+func WithProcessCommandLineOrArgsForWindows(config *runtime.ContainerConfig, image *imagespec.ImageConfig) oci.SpecOpts {
+	if image.ArgsEscaped {
+		return func(ctx context.Context, client oci.Client, c *containers.Container, s *runtimespec.Spec) (err error) {
+			// firstArgFromImg is a flag that is returned to indicate that the first arg in the slice comes from either the
+			// image Entrypoint or Cmd. If the first arg instead comes from the container config (e.g. overriding the image values),
+			// it should be false. This is done to support the non-OCI ArgsEscaped field that Docker used to determine how the image
+			// entrypoint and cmd should be interpreted.
+			//
+			args, firstArgFromImg, err := getArgs(image.Entrypoint, image.Cmd, config.GetCommand(), config.GetArgs())
+			if err != nil {
+				return err
+			}
+
+			var cmdLine string
+			if image.ArgsEscaped && firstArgFromImg {
+				cmdLine = args[0]
+				if len(args) > 1 {
+					cmdLine += " " + escapeAndCombineArgsWindows(args[1:])
+				}
+			} else {
+				cmdLine = escapeAndCombineArgsWindows(args)
+			}
+
+			return oci.WithProcessCommandLine(cmdLine)(ctx, client, c, s)
+		}
+	}
+	// if ArgsEscaped is not set
+	return func(ctx context.Context, client oci.Client, c *containers.Container, s *runtimespec.Spec) (err error) {
+		args, _, err := getArgs(image.Entrypoint, image.Cmd, config.GetCommand(), config.GetArgs())
+		if err != nil {
+			return err
+		}
+		return oci.WithProcessArgs(args...)(ctx, client, c, s)
+	}
+}
+
+// getArgs is used to evaluate the overall args for the container by taking into account the image command and entrypoints
+// along with the container command and entrypoints specified through the podspec if any
+func getArgs(imgEntrypoint []string, imgCmd []string, ctrEntrypoint []string, ctrCmd []string) ([]string, bool, error) {
+	//nolint:dupword
+	// firstArgFromImg is a flag that is returned to indicate that the first arg in the slice comes from either the image
+	// Entrypoint or Cmd. If the first arg instead comes from the container config (e.g. overriding the image values),
+	// it should be false.
+	// Essentially this means firstArgFromImg should be true iff:
+	// Ctr entrypoint	ctr cmd		image entrypoint	image cmd	firstArgFromImg
+	// --------------------------------------------------------------------------------
+	//	nil				 nil			exists			 nil		  true
+	//  nil				 nil		    nil				 exists		  true
```

Comment:
This is all oddly formatted

Reply:
(spaces/tabs seem to be screwed up)

Reply:
![image](https://user-images.githubusercontent.com/99994218/222833515-ec1cea12-757d-43ea-8270-8888778cf7ac.png)


Shows up alright on vscode for me. Not sure whats happening in the PR